pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
FWIW I'm fine with publishing this from the Kubelet. My personal opinion is that it's reasonable to think of "system node labels" and "user node labels" separately, and publish the former directly from the Kubelet rather than from a separate pod. But regardless, as @justinsb points out, it should be possible to move this back out of Kubelet later, so we don't really need to settle this debate in order to accept this PR.